lib/DDG/Meta/Data.pm

```diff
@@ -122,6 +125,26 @@ sub get_ia {
     return $m;
 }
 
+sub _satisfy {
+	my ($ia, $by, $lookup) = @_;
+	return $by->($ia, $lookup) if ref $by eq 'CODE';
+	my $pby = $ia->{$by};
+	ref $lookup eq 'CODE' ? $lookup->($pby) : $pby eq $lookup;
+}
+
+# get_ias(repo => 'goodies', dev_milestone => 'live'...)
+# Lookups combine as an AND operation.
+sub get_ias {
+	my (@lookups) = @_;
+	my @results = @ia_container;
+	foreach (pairs @lookups) {
```

Comment:
Let's keep this very simple.  We are likely to have a more flexible metadata API in the near future that would make anything complex here largely obsolete.  However, basic filtering would be useful immediately in a several places so I think it's worth entertaining.

Most of the queries are going to be similar to the SQL in clause, e.g. give me all of the IAs where repo in (one, of, these) and production_state in (this, or, that).  Let's not go down the AND/OR distinction yet.